app/_components/SelectButton.tsx

```diff
@@ -0,0 +1,42 @@
+import clsx from 'clsx';
+import { FC, SVGProps } from 'react';
+
+import styles from './SelectButton.module.scss';
+
+interface SelectButtonProps {
+  variant: 'simple' | 'detailed'; // 버튼 스타일 구분
+  title: string;
+  description?: string;
+  subTitles?: string[];
+  Icon?: FC<SVGProps<SVGSVGElement>>; // SVG 컴포넌트 타입
+  isSelected?: boolean;
+  onClick?: () => void;
+}
+
+export default function SelectButton({
+  variant = 'simple',
+  title,
+  description,
+  subTitles,
+  Icon,
+  isSelected = false,
+  onClick,
+}: SelectButtonProps) {
+  return (
+    <div className={clsx(styles.wrapper, { [styles['wrapper--selected']]: isSelected })} onClick={onClick}>
+      <div className={styles['wrapper__title']}>
+        {/* {variant === 'detailed' && Icon && <Icon />} //TODO: SVGR설정하여 아이콘 처리*/}
+        <div>{title}</div>
+      </div>
+
+      {variant === 'detailed' && (
+        <>
+          <div className={styles['wrapper__description']}>{description}</div>
+          <div className={styles['wrapper__sub-title']}>
+            {subTitles && subTitles.length > 0 && subTitles?.map((e, idx) => <span key={`subTitle-${idx}`}>{e}</span>)}
```

Comment:
1. nullish 검증이 완료된 변수에는 optional chaining을 할 필요가 없어요
2. index로 key를 구분할 때 컴포넌트의 순서가 바뀌었을 때 리액트가 컴포넌트를 구분을 못하고 불필요한 리렌더링을 할 수 있어요. 유니크한 이름을 키에 넣어주는게 나을 것 같아요
3. `e` 라는 이름이 한눈에 봤을 때 의미를 바로 알아차리기 힘들었던 것 같아요

```suggestion
            {subTitles.length > 0 && subTitles.map((subTitle) => <span key={subTitle}>{subTitle}</span>)}
```

Reply:
이런부분을 놓치고있었다니~~~~~~~